### DISCUSSION TITLE
TraceId is empty in log UI

### DISCUSSION THREAD

**Discussion by luojie233** [apache/skywalking#8124] [Q&A]
First, environment:
skywalking 8.8.1
agent 8.8.0

when I use grpc to send my log4j2 log to SkyWalking OAP server, I found traceID not attached automatically.

my pom.xml
![image](https://user-images.githubusercontent.com/17927730/141604931-5a275059-9e7a-4863-8514-1a0f964d0cf1.png)

log4j2.xml is:
![image](https://user-images.githubusercontent.com/17927730/141604782-037179c7-73c3-448a-acf1-49a10db01359.png)

OAP server shows:
![image](https://user-images.githubusercontent.com/17927730/141604871-542fedb7-6b0b-4b1b-9ee2-688a98f98be9.png)

**Comment by wu-sheng** [comment]
The key point is whether these logs are in tracing context. Not all logs have tracing context.

**Reply by luojie233** [reply]
my console log like this, all have trace_id
![image](https://user-images.githubusercontent.com/17927730/141605472-f7ae4bcb-87e8-417f-b031-408e9e183f18.png)

**Reply by wu-sheng** [reply]
Trace id printing and log reporting are different features. If one of it works, recheck your env with doc.
I assume there are some configuration issues.

**Reply by luojie233** [reply]
![image](https://user-images.githubusercontent.com/17927730/141755542-85cfa291-9694-4710-a611-5abf3a0ae5a0.png)

I changed AsyncRoot to Root, it's OK.

![image](https://user-images.githubusercontent.com/17927730/141758719-071d26bd-a79c-4320-be4f-a575dc26359e.png)

**Reply by wu-sheng** [reply]
Yes, that could be. SkyWalking reporter(log) is already async in kernel level. You don't need to rely on log4j(2) async log. Because we need the tracing context, which can be accessed only in the same thread.
